packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -1557,13 +1557,34 @@ export class ThinEngine {
     public stopRenderLoop(renderFunction?: () => void): void {
         if (!renderFunction) {
             this._activeRenderLoops.length = 0;
+            this._cancelFrame();
             return;
         }
 
         const index = this._activeRenderLoops.indexOf(renderFunction);
 
         if (index >= 0) {
             this._activeRenderLoops.splice(index, 1);
+            if (this._activeRenderLoops.length == 0) {
+                this._cancelFrame();
+            }
+        }
+    }
+
+    protected _cancelFrame() {
+        if (this._renderingQueueLaunched && this._frameHandler) {
+            this._renderingQueueLaunched = false;
+            if (!IsWindowObjectExist()) {
+                if (typeof cancelAnimationFrame === "function") {

```

Comment:
where is `canvelAnimationFrame defined, if no window object exists? in the global scope?
If yes, then why don't we just use globalThis instead of window, unless this.getHostWindow returns something that can be used? You can save this if/else

Reply:
Discussed offline